tests-perf/locust/locust-notifications.py

```diff
@@ -1,24 +1,102 @@
+
+    def __init__(self, *args, **kwargs):
+        super(NotifyApiUser, self).__init__(*args, **kwargs)
+
+        self.headers = {"Authorization": os.getenv("TEST_AUTH_HEADER")}
+        self.email = os.getenv("LOAD_TEST_EMAIL", "success@simulator.amazonses.com")
+        self.phone_number = os.getenv("LOAD_TEST_PHONE_NUMBER", "12897682684")
+        self.template_group = NotifyApiUserTemplateGroup(
+            bulk_email_id=os.getenv("LOAD_TEST_BULK_EMAIL_TEMPLATE_ID", "5ebee3b7-63c0-4052-a8cb-387b818df627"),
+            email_id=os.getenv("LOAD_TEST_EMAIL_TEMPLATE_ID", "a59b313d-8de2-4973-ac2f-66de7ec0b239"),
+            email_with_attachment_id=os.getenv("LOAD_TEST_EMAIL_WITH_ATTACHMENT_TEMPLATE_ID", "a59b313d-8de2-4973-ac2f-66de7ec0b239"),
+            email_with_link_id=os.getenv("LOAD_TEST_EMAIL_WITH_LINK_TEMPLATE_ID", "5ebee3b7-63c0-4052-a8cb-387b818df627"),
+            sms_id=os.getenv("LOAD_TEST_SMS_TEMPLATE_ID", "83d01f06-a818-4134-bd69-ce90a2949280"),
+        )
+
+    @task(16)
+    def send_email_notifications(self):
+        json = self.__email_json(self.template_group.email_id)
+
+        self.client.post("/v2/notifications/email", json=json, headers=self.headers)
+
+    @task(8)
+    def send_email_with_attachment_notifications(self):
+        personalisation = {
+            "attached_file": {
+                "file": "Q29udGVudCBvZiBBdHRhY2hlZCBmaWxl",
+                "filename": "attached_file.txt",
+                "sending_method": "attach",
+            }
+        }
+        json = self.__email_json(self.template_group.email_with_attachment_id, personalisation)
+
+        self.client.post("/v2/notifications/email", json=json, headers=self.headers)
 
-    @task
-    def send_notifications(self):
-        headers = {"Authorization": AUTH_HEADER}
+    @task(4)
+    def send_email_with_link_notifications(self):
+        personalisation = {
+            "application_file": {
+                "file": "Q29udGVudCBvZiBBdHRhY2hlZCBmaWxl",
+                "filename": "attached_file.txt",
+                "sending_method": "link",
+            }
+        }
+        json = self.__email_json(self.template_group.email_with_link_id, personalisation)
+
+        self.client.post("/v2/notifications/email", json=json, headers=self.headers)
+
+    @task(2)
+    def send_bulk_notifications(self):
         json = {
-            "email_address": "success@simulator.amazonses.com",
-            "template_id": "9c17633c-126a-4ad3-ad2f-b14c3a85314a",
-            "personalisation": {"colour": "Fulvous"},
+            "name": f"My bulk name {datetime.utcnow().isoformat()}",
+            "template_id": self.template_group.bulk_email_id,
+            "csv": rows_to_csv([["email address", "application_file"], *job_line(self.email, 2)])
         }
 
-        self.client.post("/v2/notifications/email", json=json, headers=headers)
+        self.client.post("/v2/notifications/bulk", json=json, headers=self.headers)
+
+    @task(1)
```

Comment:
Can we give a bigger weight to the SMS? something bigger than email with files but a bit lower than email with no attachment test?

Reply:
Increased it to 10.. Here are the results 

```
Name                                                          # reqs      # fails  |     Avg     Min     Max  Median  |   req/s failures/s
--------------------------------------------------------------------------------------------------------------------------------------------
 POST /v2/notifications/bulk                                        5     0(0.00%)  |     614     572     669     620  |    0.47    0.00
 POST /v2/notifications/email                                      68     0(0.00%)  |     466     257     727     450  |    6.43    0.00
 POST /v2/notifications/sms                                        27     0(0.00%)  |     405     213     491     410  |    2.55    0.00
--------------------------------------------------------------------------------------------------------------------------------------------
```